xbmc/addons/settings/AddonSettings.cpp

```diff
@@ -433,6 +433,15 @@ bool CAddonSettings::HasSettings() const
   return IsInitialized() && GetSettingsManager()->HasSettings();
 }
 
+bool CAddonSettings::Save()
+{
+  if (m_addon)
+  {
+    m_addon->SaveSettings();
+  }
+  return true;
```

Comment:
This function will always return true, should it return false if m_addon is nullptr?

Reply:
I think I'd rather change CAddon::SaveSettings() to return the actual success of the operation (there are several steps there that return a status code that is simply being ignored nowadays). If m_addon is nulll something must be wrong and the issue is elsewhere. I'd probably add an assert there and let it crash...

Reply:
done in https://github.com/xbmc/xbmc/pull/23558/commits/88a4c418ab36abe37d466c22c995d7249c714837